src/mapping/NearestNeighborGradientMapping.cpp

```diff
@@ -20,6 +20,8 @@ NearestNeighborGradientMapping::NearestNeighborGradientMapping(
     int        dimensions)
     : NearestNeighborBaseMapping(constraint, dimensions, true, "NearestNeighborGradientMapping", "nng")
 {
+  PRECICE_CHECK(!hasConstraint(CONSERVATIVE), "Nearest Neighbor Gradient is not defined for conservative mapping");
```

Comment:
```suggestion
  PRECICE_CHECK(!hasConstraint(CONSERVATIVE), "Nearest-neighbor-gradient mapping is not implemented using a \"conservative\" constraint. Please select constraint=\""consistent\" or a different mapping method.");
```
Is scaled consistent working? If it was just never tested we should also safeguard against it and explain that it requires some additional testing.

Reply:
The scaled consistent mapping is defined in the parent class 'Mapping' so I assumed that it would work exactly the same for NN, NP and NNG for now. But I can add a comment to clearify this.